### ISSUE TITLE
"An unexpected error occurred" in Firefox 71

### ISSUE THREAD

**Issue by okias** [EFForg/privacybadger#2474]
Only this message appear after disabling & enabling extension:

```
1570472797925	addons.webextension.jid1-MnnxcxisBPnSXQ-eff@jetpack	WARN	Loading extension 'jid1-MnnxcxisBPnSXQ-eff@jetpack': Reading manifest: Warning processing storage: An unexpected property was found in the WebExtension manifest.
```

**Comment by ghostwords** [comment]
Hello! I'm unable to reproduce the problem in the latest Nightly (2019-10-07). The "unexpected property" message is a harmless warning, it doesn't really mean anything.

**Comment by okias** [comment]
I get this for every page 
![Snímek z 2019-10-07 20-51-49](https://user-images.githubusercontent.com/2457435/66339923-7286de00-e944-11e9-82a7-dbebf745fa15.png)

**Comment by ghostwords** [comment]
According to https://bugzilla.mozilla.org/show_bug.cgi?id=1555491#c18, "An unexpected error occurred" has to do with failing to write to extension storage (powered by IndexedDB). This is probably not a Privacy Badger-specific problem, and it may be specific to your environment (is your Firefox partition out of space?).

**Comment by Teuniz** [comment]
I have exactly the same problem after installing FF 71 on Linux.
I tried disabling my other addons and restarted FF but the problem persists.
It worked fine with FF 70. The FF partition has 16GB of freespace.

**Comment by grdryn** [comment]
> I have exactly the same problem after installing FF 71 on Linux.

Same, specifically Fedora 31, with the latest upgrade: `firefox-71.0-8.npgo.fc31.x86_64`. I haven't tried any other distributions. Is that the same as you @Teuniz? I doubt it's an issue with the Fedora build of Firefox, but if that happens to be the same for both of us, then it's maybe worth not ruling out.

I've also noticed other extensions broken, such as simple-tab-groups, which I've reported here: https://github.com/Drive4ik/simple-tab-groups/issues/501

**Comment by Teuniz** [comment]
> > I have exactly the same problem after installing FF 71 on Linux.
> 
> Same, specifically Fedora 31, with the latest upgrade: `firefox-71.0-8.npgo.fc31.x86_64`. I haven't tried any other distributions. Is that the same as you @Teuniz? I doubt it's an issue with the Fedora build of Firefox, but if that happens to be the same for both of us, then it's maybe worth not ruling out.
> 
> I've also noticed other extensions broken, such as simple-tab-groups, which I've reported here: [Drive4ik/simple-tab-groups#501](https://github.com/Drive4ik/simple-tab-groups/issues/501)

I'm using openSUSE Leap 15.1 and FF is installed from the openSUSE Mozilla repo.
So, It doesn't look like a specific build issue from just one distro...

**Comment by ghostwords** [comment]
Could you file a [Mozilla Bugzilla](https://bugzilla.mozilla.org/) bug and link to it here? Thank you!

**Comment by chris24k** [comment]
> Same, specifically Fedora 31, with the latest upgrade: `firefox-71.0-8.npgo.fc31.x86_64`. I haven't tried any other distributions. Is that the same as you @Teuniz? I doubt it's an issue with the Fedora build of Firefox, but if that happens to be the same for both of us, then it's maybe worth not ruling out.
> 
> I've also noticed other extensions broken, such as simple-tab-groups, which I've reported here: [Drive4ik/simple-tab-groups#501](https://github.com/Drive4ik/simple-tab-groups/issues/501)

Also experiencing this error on Fedora 31, along with some other weird behavior (such as extensions not saving settings). I reinstalled FF71 and it seems to have fixed the settings problem, but still getting "An unexpected error occurred" in Privacy Badger.

**Comment by ghostwords** [comment]
>still getting "An unexpected error occurred" in Privacy Badger

Privacy Badger reports [browser errors that come up when trying to write to extension storage](https://github.com/EFForg/privacybadger/blob/66783e4c999d4ed6e039a5e34d866fe97f4a8df2/src/js/storage.js#L608-L617), other extensions probably do not (and fail silently).

**Comment by linkp** [comment]
[keepassxreboot/keepassxc-browser](https://github.com/keepassxreboot/keepassxc-browser) is experiencing similar issues and throws an error to the console: "Error: Cannot load extension settings"

I noticed both immediately following the upgrade to Firefox 71 on OpenSUSE Leap 15.1 yesterday.

**Comment by chris24k** [comment]
> [keepassxreboot/keepassxc-browser](https://github.com/keepassxreboot/keepassxc-browser) is experiencing similar issues and throws an error to the console: "Error: Cannot load extension settings"
> 
> I noticed both immediately following the upgrade to Firefox 71 on OpenSUSE Leap 15.1 yesterday.

Yea same problem here. The settings panel doesn't load so I can't connect to my database. Also, uBlock Origin deletes its filters and settings whenever I close Firefox.  I'm guessing this is a problem with Fedora's FF build.

**Comment by chris24k** [comment]
I submitted a report on Mozilla Bugzilla for those with the same problem: https://bugzilla.mozilla.org/show_bug.cgi?id=1601424

**Comment by alex65536** [comment]
Just upgraded to 71.0 from Debian unstable and got the same problem.

**Comment by antistress** [comment]
> Just upgraded to 71.0 from Debian unstable and got the same problem.

Just like me

**Comment by Stupendoussteve** [comment]
Seems to work fine in the Firefox 71.0 build provided by Mozilla, on Fedora. The Fedora-built one does appear to be broken (Fedora bug https://bugzilla.redhat.com/show_bug.cgi?id=1779570).

**Comment by bcl** [comment]
Downgrading Fedora 31 to firefox-69.0.1-3.fc31.x86_64 also works fine.

**Comment by xfix** [comment]
Also happens for me on NixOS 19.09.

**Comment by Teuniz** [comment]
For me the problem seems to be resolved after upgrading FF to 71.0-lp151.5.1 from the openSUSE repo.

**Comment by alex65536** [comment]
Debian also fixed the bug in firefox 71.0-2.
Maybe the issue can be closed, because it's not a bug of the extension itself?

**Comment by ghostwords** [comment]
Great news! I'll close this once new reports stop coming in.

**Comment by das-g** [comment]
> Also happens for me on NixOS 19.09. KeepassXC Browser extension doesn't show settings page. uBlock Origin shows suspiciously low number of blocked advertisements since installing it (that lowers when restarting a web browser), Decentraleyes always shows 1 locally injected resource.
> 
> I think everything is broken in Firefox 71.

I had this problem, too and just now upgraded to `nixos-19.09.1549.45ea6092203` and the problem seems to be gone. :tada:

**Comment by LorbusChris** [comment]
Fixed on Fedora with `firefox-71.0-15.fc31` as well

**Comment by renatobellotti** [comment]
Fixed on Fedora 30 as well.

**Comment by go2null** [comment]
fixed with `firefox-71.0_2` on Void Linux

**Comment by ghostwords** [comment]
Haven't gotten any new reports in a while, resolving.
